rfd/XXXX-tsh-upgrade.md

```diff
@@ -0,0 +1,117 @@
+---
+authors: Bernard Kim (bernard@goteleport.com)
+state: draft
+---
+# RFD XXXX - tsh upgrade
+
+## Required Approvers
+
+* Engineering: ...
+* Infra: ...
+* Kube: ...
+
+## What
+
+A tsh subcommand that will prompt the user to upgrade if the version of tsh is older than the cluster version.
+
+## Why
+
+One of our goals for Teleport Cloud is to increase the cadence at which clusters can be upgraded.
+This requires that Teleport Agents and Teleport client tools be up to date with the latest compatible version of the cluster.
+
+Manually upgrading Teleport Agents and client tools can be a tedious task, so customers can end up delaying their upgrades.
+Simplifying this process will help customers upgrade and stay up to date with their Teleport version.
+
+Development for Automatic Upgrades of Teleport Agents has already been completed, so the success criteria for this RFD only includes developing a simpler way to keep Teleport client tools up to date.
+
+## Details
+
+The main idea here is to add an upgrade command to the Teleport client tools. This will detect the version of the Teleport cluster.
+If the version of tsh is older than the version of the cluster, the user will be prompted to upgrade.
```

Comment:
As we discussed in our sync, I think the criteria should be "if tsh/tctl version is different from the cluster it's talking to" so it would upgrade/downgrade itself as needed.